src/System.Slices/src/System/Span.cs

```diff
@@ -283,45 +288,25 @@ public bool ReferenceEquals(Span<T> other)
 
         public override int GetHashCode()
         {
-            // TODO: Write something better
-            uint hash = unchecked((uint)Length);
-            foreach (T el in this)
+            var hashCode = Length ^ Offset.GetHashCode();
```

Comment:
@ellismg, could you comment on the hashcode?

Since we implement IEquatable, we should override GetHashCode. But I have to say using spans as hash-table keys is something we should not encourage, especially that we might need to make Span<T> a stackonly type.